woocommerce/woocommerce
# fix: Typescript issues in preparation for pnpm 7 upgrade

One of the blockers for PNPM 7 upgrade is the change in `public-hoist-pattern` as described in #33685.

This pull request addresses the missing Typescript definition packages across the packages in the project as well as WooCommerce core.


### All Submissions:






### Changes proposed in this Pull Request:

Most of the changes are just additions of `@types/*`

It doesn't seem like `eslint` will be an issue as we run eslint through our own `@woocommerce/eslint-plugin` in the workspace.

### How to test the changes in this Pull Request:

As this PR does not complete the PNPM 7 migration, in order to test this you will need to install PNPM 7 and update the 'engines' field in `plugins/woocommerce-admin/package.json`, `plugins/woocommerce/package.json` and `tools/cherry-pick/package.json` to allow PNPM 7.

Aside from that, you will also need to install PNPM 7: `npm install -g pnpm@7.8.0`

Once those steps have been completed,

1. `pnpm install`
2. `pnpm -- turbo run --no-cache build --filter=woocommerce`

With PNPM v7 you will see the error below, which can be fixed by removing the two `// @ts-expect-error` comments. 

Though I'm not entirely sure why the error goes away with PNPM v7, it doesn't seem quite related and will require a migration of the component to TS to resolve properly.

![image](https://user-images.githubusercontent.com/27843274/182557807-a14a269f-ab8e-4e4e-baca-ffe28b96f3cc.png)

Once these errors are removed there should be 0 errors in the build process.

### Other information:




### FOR PR REVIEWER ONLY:
